model/utils.go

```diff
@@ -141,26 +141,17 @@ func NewRandomString(length int) string {
 	return b.String()
 }
 
-// GetMillis is a convience method to get milliseconds since epoch.
+// GetMillis is a convenience method to get milliseconds since epoch.
 func GetMillis() int64 {
 	return time.Now().UnixNano() / int64(time.Millisecond)
 }
 
-// GetMillisForTime is a convience method to get milliseconds since epoch for provided Time.
+// GetMillisForTime is a convenience method to get milliseconds since epoch for provided Time.
 func GetMillisForTime(thisTime time.Time) int64 {
 	return thisTime.UnixNano() / int64(time.Millisecond)
 }
 
-// ParseDateFilterToTime is a convience method to get Time from string
-func ParseDateFilterToTime(filterString string) time.Time {
```

Comment:
I still see the value of have this function to reduce the code repetition. Maybe we can use it and change to simply:
`return time.Parse("2006-01-02", PadDateStringZeros(filterString))` and handling the error properly in each case. Because the function itself is the same, what is changing is the error handling, so move the error handling outside and keep the common behavior inside the function.

Reply:
I haven't a strong opinion here.